include/FileBrowser.h

```diff
@@ -199,16 +182,22 @@ private slots:
 	void updateDirectory( QTreeWidgetItem * item );
 } ;
 
+class FileBrowserWidgetItem : public QTreeWidgetItem
+{
+public:
+	FileBrowserWidgetItem(const QStringList& strings, int type, QTreeWidget* parent = nullptr);
+	virtual QString fullName(QString path = QString{}) const = 0;
+};
 
 
-class Directory : public QTreeWidgetItem
+class Directory : public FileBrowserWidgetItem
 {
 public:
-	Directory(const QString& filename, const QString& path, const QString& filter, bool disableEntryPopulation = false);
+	Directory(const QString& filename, const QString& path, const QString& filter);
 
 	void update();
 
-	inline QString fullName( QString path = QString() )
+	QString fullName(QString path = QString{})  const override
```

Comment:
```suggestion
	QString fullName(QString path) const override
```
I could be wrong, but the default argument shouldn't be needed when it's already present in the base class method.